pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -341,6 +341,80 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 	return nil
 }
 
+// determineExtraMonToRemove assumes all mons are in quorum and that there are more mons
+// that required for desired state. One mon will be picked for removal in this priority:
+// 1. If a stretch cluster, remove the extra mon according to the stretch topology
+// 2. If more than one mon on a node, remove one of them
+// 3. If no criteria require for 1 or 2, pick an arbitrary mon
+func (c *Cluster) determineExtraMonToRemove() string {
+	mons := c.clusterInfoToMonConfig()
+	if c.spec.IsStretchCluster() {
+		stretchMonToRemove := c.findExtraMonToRemoveFromStretchCluster(mons)
+		if stretchMonToRemove != "" {
+			return stretchMonToRemove
+		}
+		logger.Infof("did not find an extra mon to remove from the stretch cluster")
+		return ""
+	}
+
+	nodesWithMons := map[string]string{}
+	arbitraryMon := ""
+	for _, mon := range mons {
+		if mon.NodeName == "" {
+			logger.Infof("mon %q is not assigned a host yet", mon.DaemonName)
+			continue
+		}
+		// Check if there are multiple mons on the node
+		if existingMon, ok := nodesWithMons[mon.NodeName]; ok {
+			logger.Infof("found mons %q and %q on node %s, removing mon %q", existingMon, mon.DaemonName, mon.NodeName, mon.DaemonName)
+			return mon.DaemonName
+		}
+		nodesWithMons[mon.NodeName] = mon.DaemonName
+
+		// assign the current mon as the fallback mon
+		arbitraryMon = mon.DaemonName
+	}
+
+	logger.Infof("removing arbitrary extra mon %q", arbitraryMon)
+	return arbitraryMon
+}
+
+func (c *Cluster) findExtraMonToRemoveFromStretchCluster(mons []*monConfig) string {
+	// Build the count of current mons per zone
+	zoneCount := map[string]int{}
+	monInZones := map[string]string{}
+	for _, m := range mons {
+		if m.Zone == "" {
+			logger.Warningf("zone not found on mon %q", m.DaemonName)
+			continue
+		}
+		zoneCount[m.Zone]++
+		// We just need the name of one of the mons in the zone in case there are extra
+		monInZones[m.Zone] = m.DaemonName
+	}
+
+	// Find a zone that has too many mons
+	for _, zone := range c.spec.Mon.StretchCluster.Zones {
+		count, ok := zoneCount[zone.Name]
+		if !ok {
+			// The zone isn't currently assigned to any mon, so skip it
+			continue
+		}
+		if zone.Arbiter {
+			if count > 1 {
+				logger.Infof("removing extra mon %q in arbiter zone %q", monInZones[zone.Name], zone.Name)
+				return monInZones[zone.Name]
+			}
+		} else {
+			if count > 2 {
+				logger.Infof("removing extra mon %q in zone %q", monInZones[zone.Name], zone.Name)
+				return monInZones[zone.Name]
+			}
+		}
+	}
+	return ""
```

Comment:
Rather than an error, the mon failover request will be skipped [here](https://github.com/rook/rook/pull/10717/files#diff-e4d155fcec693e23523a6f0bcbeb7a341e5c8e1b5d2c6fcaa38eab93826f31dfR631). It's essentially treated as an error condition, so I don't think we need an explicit error.